### ISSUE TITLE
Verify Kernel Routing

### ISSUE THREAD

**Issue by Eupham** [Eupham/KernelDev#99]
I need you to verify that model.py for both tasks (teacher forcing and cocktail party) are being appropriately communicated to original_kernel.py
Don't stop there.  I need to know if once inside of original_kernel.py they are following the appropriate route given what was communicated.  I need what was communicated to use all of the special tokens allotted to those specific tasks.

Let's not use attention mask from model, instead lets just use metadata to build the attention behavior inside the kernel.  Let's not make the kernel route based off task or mask but based off tokens specified in metadata alone.

**Comment by Eupham** [comment]
@copilot

This fix created nans. Specifically address only the nan issues

=== Initial Evaluation ===
Evaluation for task teacher_forcing limited to 10 batches for speed
Evaluation for task cocktail_party limited to 10 batches for speed
Evaluation for task teacher_forcing limited to 10 batches for speed
Evaluation for task cocktail_party limited to 10 batches for speed
Initial training loss: nan
Initial validation loss: nan
Initial cocktail party metrics: accuracy: 0.3107,

=== Starting Training ===
Starting training for 1 epochs...
Calculated steps_per_epoch: 1000
Scheduler initialized with T_max = 1000 for a single decay.
Evaluation for task teacher_forcing limited to 50 batches for speed
Evaluation for task cocktail_party limited to 50 batches for speed
Initial validation loss: nan
Initial cocktail party metrics: accuracy: 0.3189,
/usr/local/lib/python3.12/site-packages/torch/optim/lr_scheduler.py:192: UserWarning: Detected call of lr_scheduler.step() before optimizer.step(). In PyTorch 1.1.0 and later, you should call them in the opposite order: optimizer.step() before lr_scheduler.step(). Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
